listenbrainz/webserver/views/playlist_api.py

```diff
@@ -0,0 +1,516 @@
+    https://musicbrainz.org/doc/jspf .
+
+    If the offset is provided in the URL, then the recordings will be added at that offset,
+    otherwise they will be added at the end of the playlist.
+
+    You may only add :data:`~webserver.views.playlist_api.MAX_RECORDINGS_PER_ADD` recordings in one
+    call to this endpoint.
+
+    :reqheader Authorization: Token <user token>
+    :statuscode 200: playlist accepted.
+    :statuscode 400: invalid JSON sent, see error message for details.
+    :statuscode 401: invalid authorization. See error message for details.
+    :statuscode 403: forbidden. the requesting user was not allowed to carry out this operation.
+    :resheader Content-Type: *application/json*
+    """
+
+    user = _validate_auth_header()
+    if offset is not None and offset < 0:
+        log_raise_400("Offset must be a positive integer.")
+
+    if not is_valid_uuid(playlist_mbid):
+        log_raise_400("Provided playlist ID is invalid.")
+
+    playlist = db_playlist.get_by_mbid(playlist_mbid)
+    if playlist is None or \
+       (playlist.creator_id != user["id"] and not playlist.public):
+        raise APINotFound("Cannot find playlist: %s" % playlist_mbid)
+
+    if playlist.creator_id != user["id"]:
+        raise APIForbidden("You are not allowed to add recordings to this playlist.")
+
+    data = request.json
+    validate_playlist(data, False)
+
+    if len(data["playlist"]["track"]) > MAX_RECORDINGS_PER_ADD:
+        log_raise_400("You may only add max %d recordings per call." % MAX_RECORDINGS_PER_ADD)
+
+    precordings = []
+    if "track" in data["playlist"]:
+        for track in data["playlist"]["track"]:
+            try:
+                mbid = UUID(track['identifier'][len(PLAYLIST_TRACK_URI_PREFIX):])
+            except (KeyError, ValueError):
+                log_raise_400("Track %d has an invalid identifier field, it must be a complete URI.")
+            precordings.append(WritablePlaylistRecording(mbid=mbid, added_by_id=user["id"]))
+
+    try:
+        db_playlist.add_recordings_to_playlist(playlist, precordings, offset)
+    except Exception as e:
+        current_app.logger.error("Error while adding recordings to playlist: {}".format(e))
+        raise APIInternalServerError("Failed to add recordings to the playlist. Please try again.")
+
+    return jsonify({'status': 'ok'})
+
+
+@playlist_api_bp.route("/<playlist_mbid>/item/move", methods=["POST", "OPTIONS"])
+@crossdomain(headers="Authorization, Content-Type")
+@ratelimit()
+def move_playlist_item(playlist_mbid):
+    """
+
+    To move an item in a playlist, the POST data needs to specify the recording MBID and current index
+    of the track to move (from), where to move it to (to) and how many tracks from that position should
+    be moved (count). The format of the post data should look as follows:
+
+     {“mbid” : “<mbid>”, “from” : 3, “to” : 4, “count”: 2}
```

Comment:
We can use sphinx's `code-block` here: https://www.sphinx-doc.org/en/1.5/markup/code.html

See how it's done in CB: https://github.com/metabrainz/critiquebrainz/blob/75ddff3569036bb7296097d0b61d90684581dd74/critiquebrainz/ws/review/views.py#L41-L50

Make sure we use boring quotes instead of magic smart ones!